### DISCUSSION TITLE
"remote error: tls: unrecognized name" when adding git repository

### DISCUSSION THREAD

**Discussion by tgkul** [argoproj/argo-cd#20597] [Q&A]
Hi, I recently setup my Kubernetes cluster and deployed Rancher, Cert Manager and then Argo CD. I have a Next JS web app that I want to deploy to my Kubernetes cluster using Argo CD.

The problem I am facing is with adding git repository in Argo CD. For both public and private repositories on Github and Gitlab, I get following error -

```
> argocd repo add https://gitlab.com/gitlab-org/gitlab.git --loglevel debug --logformat json

{"level":"warning","msg":"Failed to invoke grpc call. Use flag --grpc-web in grpc calls. To avoid this warning message, use flag --grpc-web.","time":"2024-10-30T18:58:55+05:30"}
{"level":"fatal","msg":"rpc error: code = Unknown desc = error testing repository connectivity: Get \"https://gitlab.com/gitlab-org/gitlab.git/info/refs?service=git-upload-pack\": remote error: tls: unrecognized name","time":"2024-10-30T18:58:55+05:30"}
```

I can get rid of grpc call error by using --grpc-web flag. It's the second error I am not able to fix.

Here is my setup:

<img width="802" alt="image" src="https://github.com/user-attachments/assets/1a274914-126f-4833-952f-e37d81965c02">

Just out of curiosity I tried running git clone on argocd-server pod but it failed for `gnutls_handshake() failed: The server name sent was not recognized`. I am not even sure if this is supposed to succeed or fail.

```
argocd@argocd-server-78c87b9bb6-4v2fc:~$ git clone https://github.com/argoproj/argo-cd.git
Cloning into 'argo-cd'...
fatal: unable to access 'https://github.com/argoproj/argo-cd.git/': gnutls_handshake() failed: The server name sent was not recognized
```

Can someone help me figure out what I am doing wrong? Let me know if you need more information. I am running out of options.

**Comment by onno204** [comment]
Hi,
Did you ever find out how to fix this issue? I have the same issue

**Reply by onno204** [reply]
I solved the problem finally.

I deployed ArgoCD using helm, which allowed me to set [repoServer.dnsConfig ](https://github.com/argoproj/argo-helm/blob/6b1ea92acba9d15f423cc03efc4d7a18dee6f9b5/charts/argo-cd/values.yaml#L2632) to 
```
  dnsConfig:
    nameservers:
      - 1.1.1.1
      - 8.8.8.8
      - 8.8.4.4
```
And 2 lines below set `dnsPolicy: "None"`.

After applying, and restarting the argocd-repo-server container the TLS unrecognized name issue was resolved instantly

**Comment by ethanbayliss** [comment]
I found out this issue was related to my wildcard domain name pointing to the same domain as my DNS suffix of the hosts.

- In my network DNS adds the `<host>.domain.com` DNS suffix to every DHCP lease
- Hosts inside of the network resolve `google.com` > `google.com.domain.com` if the suffixed `.` is not appended like `google.com.`
- I have also setup `*.domain.com` > resolves to my public IP
- That means if you try to fetch `github.com` from inside my network you may get `github.com.domain.com`

To fix this I removed my wildcard DNS entry.
See https://forum.opnsense.org/index.php?topic=37933.0
